src/main/java/core/basesyntax/service/strategy/BalanceFruitTransactionHandler.java

```
@@ -0,0 +1,17 @@
+package core.basesyntax.service.strategy;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+
+public class BalanceFruitTransactionHandler implements FruitTransactionHandler {
+    private final Storage storage;
+
+    public BalanceFruitTransactionHandler(Storage storage) {
+        this.storage = storage;

```

Constructor accepts a Storage dependency but does not validate it. Per the project guidance, public service constructors should fail fast on null dependencies to improve observability and testability. Add a null check and throw an informative RuntimeException if `storage` is null (for example, `throw new RuntimeException("Storage must not be null")`). This follows the general rule to check null parameters.